pennylane_lightning/lightning_tensor/_tensornet.py

```diff
@@ -21,12 +21,52 @@
 except ImportError:
     pass
 
+from itertools import product
 
 import numpy as np
 import pennylane as qml
 from pennylane import BasisState, DeviceError, StatePrep
 from pennylane.ops.op_math import Adjoint
 from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+
+def svd_split(M, bond_dim):
+    """SVD split a matrix into a matrix product state via numpy linalg. Note that this function is to be moved to the C++ layer."""
+    U, S, Vd = np.linalg.svd(M, full_matrices=False)
+    U = U @ np.diag(S)  # Append singular values to U
+    bonds = len(S)
+    Vd = Vd.reshape(bonds, 2, -1)
+    U = U.reshape((-1, 2, bonds))
+
+    # keep only chi bonds
+    chi = np.min([bonds, bond_dim])
+    U, S, Vd = U[:, :, :chi], S[:chi], Vd[:chi]
+    return U, Vd
+
+
+def dense_to_mps(psi, n_wires, bond_dim):
+    """Convert a dense state vector to a matrix product state."""
+    Ms = []
```

Comment:
We could pre-allocate this `Ms = [] * n_wires`.

Reply:
Do you mean `Ms = [[] for _ in range(n_wires)]` ? `[] * n_wires` will create a empty list